Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -421,6 +498,13 @@ extension BidirectionalCollection where SubSequence == Substring {
     /// - Parameter regex: The regex to search for.
     /// - Returns: A collection of matches of `regex`.
     public func matches<R: RegexComponent>(of regex: R) -> some Collection<Regex<R.RegexOuput>.Match>
+    
+    /// Returns a collection containing all matches of the specified regex.
+    /// - Parameter content: A closure to produce a `RegexComponent` to search for.
+    /// - Returns: A collection of matches of `regex`.
+    public func matches<R: RegexComponent>(
+        @RegexComponentBuilder _ content: () -> R
```

Comment:
Shall we keep `of:`?  Even though it disappears as a trailing closure, the user may choose not to use a trailing closure for clarity, e.g. `prefix(where: { $0 != 1 })`

Reply:
good point. I wasn't sure about this. Thanks